indico/modules/events/abstracts/controllers/reviewing.py

```diff
@@ -101,18 +105,38 @@ def _process(self):
         return jsonify_data(box_html=tpl.render_decision_box(self.abstract, form, management=self.management))
 
 
-class RHResetAbstractJudgment(RHAbstractReviewBase):
+class RHResetAbstractState(RHAbstractReviewBase):
+    def _checkProtection(self):
+        if (not self.abstract.can_judge(session.user, check_state=True) and
+                not (self.abstract.state == AbstractState.withdrawn and self.event_new.can_manage(session.user))):
+            raise Forbidden
+        RHAbstractReviewBase._checkProtection(self)
+
     def _process(self):
-        if self.abstract.state not in (AbstractState.submitted, AbstractState.withdrawn):
-            reset_abstract_judgment(self.abstract)
-            flash(_("Abstract judgment has been reset"), 'success')
+        if self.abstract.state != AbstractState.submitted:
+            reset_abstract_state(self.abstract)
+            flash(_("Abstract state has been reset"), 'success')
         html = render_abstract_page(self.abstract, management=self.management)
         return jsonify_data(display_html=html, management_html=html)
 
 
-class RHReviewAbstractForTrack(RHAbstractReviewBase):
-    CSRF_ENABLED = True
+class RHWithdrawAbstract(RHAbstractReviewBase):
+    def _checkProtection(self):
+        abstract = self.abstract
+        if (not self.event_new.can_manage(session.user) and
```

Comment:
Should we have `can_withdraw()` since we probably need to chek this in the template too to know whether to show the action or not?